File: JavaScript/src/algorithm/SlidingWindow.js
const arr = [0, 1, 2, 3, 4];
const length = arr.length
const kTerm = 2

const maxSum = (array, n, k) => {
    if (n < k) {
        return 0;
    }

    let sum = 0
    for (let counter = 0; counter < k; counter++) {
        sum += array[counter]
    }
    let windowSum = sum
    for (let counter = k; counter < n; counter++) {
        const next = array[counter]
        const prev = array[counter - k]
        windowSum += next - prev
        sum = Math.max(windowSum, sum)
    }
    return sum
}

console.log(maxSum(arr, length, kTerm))

File: src/main/java/algorithm/slidingwindow/SlidingWindow.java
package algorithm.slidingwindow;

import java.util.logging.Logger;

public class SlidingWindow {
    public static void main(String[] args) {
        Logger logger = Logger.getLogger(SlidingWindow.class.getName());
        int[] arr = {0, 1, 2, 3, 4};
        int k = 2;
        int n = arr.length;
        String res = String.valueOf(maxSum(arr, n, k));
        logger.info(res);
    }

    private static int maxSum(int[] arr, int n, int k) {
        int maxSum = 0;
        if (n > k) {
            // first k terms
            for (int i = 0; i < k; i++) {
                maxSum += arr[i];
            }
            int windowSum = maxSum;
            // from k to n terms
            for (int j = k; j < n; j++) {
                int prev = arr[j - k];
                int next = arr[j];
                windowSum += next - prev;
                maxSum = Math.max(windowSum, maxSum);
            }
        }
        return maxSum;
    }

}